pvlib/pvsystem.py

```diff
@@ -1172,6 +1202,150 @@ def calcparams_desoto(effective_irradiance, temp_cell,
     return IL, I0, Rs, Rsh, nNsVth
 
 
+def calcparams_pvsyst(effective_irradiance, temp_cell,
+                      alpha_sc, gamma_ref, mugamma,
+                      I_L_ref, I_o_ref, 
+                      R_sh_ref, R_sh_0, R_s, 
+                      cells_in_series,
+                      R_sh_exp=5.5,
+                      EgRef=1.121,
+                      irrad_ref=1000, temp_ref=25):
+    '''
+    Calculates five parameter values for the single diode equation at 
+    effective irradiance and cell temperature using the PVsyst v6 
+    model described in [1,2,3]. The five values returned by calcparams_pvsyst
+    can be used by singlediode to calculate an IV curve.
+
+    Parameters
+    ----------
+    effective_irradiance : numeric
+        The irradiance (W/m2) that is converted to photocurrent.
+
+    temp_cell : numeric
+        The average cell temperature of cells within a module in C.
+
+    alpha_sc : float
+        The short-circuit current temperature coefficient of the
+        module in units of A/C.
+
+    gamma_ref : float
+        The diode ideality factor
+
+    mugamma : float
+        The temperature coefficient for the diode ideality factor, 1/K
+
+    I_L_ref : float
+        The light-generated current (or photocurrent) at reference conditions,
+        in amperes.
+        
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions,
+        in amperes.
+        
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+        
+    R_sh_0 : float
+        The shunt resistance at zero irradiance conditions, in ohms.
+        
+    R_s : float
```

Comment:
R_s_ref seems more consistent with other variables at reference conditions.

Reply:
I agree. But here we run into an unfortunate linkage to the SAM database for the CEC model parameters.  `R_s` is the heading in that file, which is used to define the keys for the module parameter dict. Although in my opinion `R_s` for PVsyst is a different parameter than `R_s` for De Soto or CEC (because parameter values are determined jointly, they are intrinsically linked and can't be regarded as correct for a different model) I do think it's useful to maintain a common nomenclature.

I think changing `R_s` to `R_s_ref` will involve some work in `pvsystem.retrieve_sam` so I'm not going to take that on in this PR.